frontend/simple/views/containers/Proposals.vue

```diff
@@ -55,138 +61,130 @@
-          }
-        }
-      ]
-    }
+    Voting
   },
   computed: {
-    groupProposals () {
+    ...mapGetters([
+      'currentUserIdentityContract',
+      'currentGroupState'
+    ]),
+    proposalList () {
       const notVoted = []
       const alreadyVoted = []
       const own = []
 
-      for (var i = 0, l = this.mockProposals.length; i < l; i++) {
-        const proposal = this.mockProposals[i]
-        if (proposal.data.ownProposal) {
-          own.push(proposal)
-        } else if (proposal.data.userVote === null) {
-          notVoted.push(proposal)
-        } else if (proposal.data.userVote !== null) {
-          alreadyVoted.push(proposal)
+      const groupData = this.currentGroupState
+      const proposals = groupData.proposals
+      const userData = this.currentUserIdentityContract.attributes
+
+      for (let hash in proposals) {
+        const proposalData = {
+          type: proposals[hash].type, // 'invitation' or 'removal' for member, field name for rule/mincome
+          votes: {
+            total: Object.entries(groupData.profiles).length,
+            received: proposals[hash].for.length + proposals[hash].against.length,
+            threshold: proposals[hash].threshold
+          },
+          value: proposals[hash].value || proposals[hash].candidate || null,
+          originalValue: groupData[proposals[hash].type] || null,
+          ownVote: proposals[hash].for.includes(userData.name) || proposals[hash].against.includes(userData.name)
+            ? proposals[hash].for.includes(userData.name)
+            : null,
+          isOwnProposal: proposals[hash].initiator === userData.name,
+          initiator: proposals[hash].initiator,
+          hash: hash
+        }
+
+        if (proposalData.isOwnProposal) {
+          own.push(proposalData)
+        } else if (proposalData.ownVote === null) {
+          notVoted.push(proposalData)
+        } else if (proposalData.ownVote !== null) {
+          alreadyVoted.push(proposalData)
         }
       }
 
       return { notVoted, alreadyVoted, own }
     },
     allVoted () {
-      const { notVoted, alreadyVoted } = this.groupProposals
+      const { notVoted, alreadyVoted } = this.proposalList
 
       return alreadyVoted.length && notVoted.length === 0
     }
   },
   methods: {
-    handleVoteAgainst () {
-      console.log('TODO Logic - The user voted against')
+    async handleVoteAgainst (hash) {
+      try {
+        // Create a vote against the proposal
+        const groupId = this.$store.state.currentGroupId
+        const latest = await backend.latestHash(groupId)
+        const vote = new Events.HashableGroupVoteAgainstProposal({
+          username: this.currentUserIdentityContract.attributes.name,
+          proposalHash: hash
+        }, latest)
+        await backend.publishLogEntry(groupId, vote)
+      } catch (ex) {
+        // TODO: diplay error to user
+        console.error('Failed to cast vote:')
+        console.log(ex)
```

Comment:
My suggestion to show the error on UI is to simple replace the computed "helperText" for "props.errorText" on Voting component when necessary. Can you do it in this MR or should be done in another PR?

Reply:
that sounds great and easy to do, will do in this one!